vowpalwabbit/core/include/vw/core/confidence_sequence.h

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) by respective owners including Yahoo!, Microsoft, and
+// individual contributors. All rights reserved. Released under a BSD (revised)
+// license as described in the file LICENSE.
+
+#pragma once
+
+#define __STDCPP_MATH_SPEC_FUNCS__ 201003L
+#define __STDCPP_WANT_MATH_SPEC_FUNCS__ 1
+
+#include "vw/core/io_buf.h"
+
+#include <algorithm>
+
+constexpr float DEFAULT_ALPHA = 0.05f;
+
+namespace VW
+{
+namespace confidence_sequence
+{
+struct IncrementalFsum
+{
+  std::vector<double> partials;
+
+  IncrementalFsum operator+=(double x)
+  {
+    int i = 0;
+    for (double y : this->partials)
+    {
+      if (std::abs(x) < std::abs(y)) { std::swap(x, y); }
+      double hi = x + y;
+      double lo = y - (hi - x);
+      if (lo != 0.0)
+      {
+        this->partials[i] = lo;
+        ++i;
+      }
+      x = hi;
+    }
+    this->partials[i] = x;
+    this->partials.resize(i + 1);
+    return *this;
```

Comment:
make a test similar to this one distributionally_robust_test.cc

Reply:
Added test